middlewares/auth.middleware.js

```diff
@@ -249,6 +272,55 @@ function deleteResetToken(req, res, next) {
     );
 }
 
+/**
+ * Middleware that creates rolebinding to access POST route for respective account
+ * @param {{body: {account:{accountType:String, id: ObjectId}}}} req the request object
+ * @param {*} res 
+ * @param {(err?)=>void} next 
+ */
+async function addCreationRoleBindings(req, res, next){
+    // Get the default role for the account type given
+    const roleName = Constants.POST_ROLES[req.body.account.accountType];
+    await Services.RoleBinding.createRoleBindingByRoleName(req.body.account.id, roleName);
+    next();
+}
+
+/**
+ * Adds proper account rolebindings on account creation
+ * @param {string} roleName name of the role to be added to account
+ */
+function addRoleBindings(roleName = undefined){
+    return async (req, res, next) => {
+        if(roleName == Constants.HACKER){
```

Comment:
I think I made my previous comment on an outdated version - whoops. Copied and pasted it here.

If we're looking for .accountId, that should be the same as the req.user.id right? Can we use that instead of searching through all the _____Details? Though using req.user.id may cause an issue if we need to create some type without having the user be logged in. Nevertheless, I'm a little hesitant about having to check through all the ____Details for creation from a maintenance and extensibility perspective.

Reply:
Also, the naming, perhaps from addRoleBindings to createRoleBindings. Made that comment on the outdated version as well.

Reply:
I don't think it would cause an issue as this method is only called when creating the subaccount which requires the user to be logged in(as the user needs permissions to POST to the route)